src/edu/ucsb/nceas/metacat/properties/PropertiesWrapper.java

```diff
@@ -0,0 +1,455 @@
+                Settings.augmentConfiguration(defaultPropertiesFilePath.toString());
+                Settings.augmentConfiguration(sitePropertiesFilePath.toString());
+            } catch (ConfigurationException e) {
+                logMetacat.error("Could not augment DataONE properties. " + e.getMessage(), e);
+            }
+
+            // mainMetaData holds configuration information about main properties. This is primarily
+            // used to display input fields on the configuration page. The information is retrieved
+            // from an xml metadata file
+            mainMetadataFilePath =
+                Paths.get(PropertyService.CONFIG_FILE_DIR, "metacat.properties.metadata.xml");
+            mainMetaData = new PropertiesMetaData(mainMetadataFilePath.toString());
+
+        } catch (TransformerException e) {
+            logAndThrow("PropertiesWrapper.initialize(): problem loading PropertiesMetaData (from "
+                + mainMetadataFilePath + ")", e);
+        } catch (IOException e) {
+            logAndThrow("PropertiesWrapper.initialize(): I/O problem while loading properties: ",
+                e);
+        } catch (MetacatUtilException e) {
+            logAndThrow("PropertiesWrapper.initialize(): SystemUtil.discoverExternalDir()", e);
+        }
+    }
+
+    private void initSitePropertiesFilePath() throws GeneralPropertyException, IOException {
+
+        if (isNotBlankPath(sitePropertiesFilePath)) {
+            logMetacat.debug("PropertiesWrapper.initSitePropertiesFilePath(): already set: "
+                + sitePropertiesFilePath);
+            if (!Files.exists(sitePropertiesFilePath)) {
+                logAndThrow("PropertiesWrapper received non-existent Site Properties File Path: "
+                    + sitePropertiesFilePath, null);
+            }
+        } else {
+            //NOTE THIS ONLY READS *DEFAULT* PROPS; SITE PROPS NOT YET LOADED/OVERLAID!!
+            // If the path has NOT been edited, it will be the default "/var/metacat/config" in
+            // metacat.properties.
+            // If the path HAS been edited, it should have been written through to the default
+            // properties, to overwrite the original value - so we should be OK either way...
+            sitePropertiesFilePath =
+                Paths.get(getProperty(PropertyService.SITE_PROPERTIES_DIR_PATH_KEY),
+                    PropertyService.SITE_PROPERTIES_FILENAME);
+
+            if (!Files.exists(sitePropertiesFilePath)) {
+                //EITHER:
+                // (1) it's the first run, so there are no site props, OR
+                // (2) the site props exists and has already been stored at a path different from
+                // the default, but this is a re-installation that has overwritten the changes in
+                // metacat.properties
+
+                // For both cases, create an empty file, since none exists at this path:
+                Files.createDirectories(
```

Comment:
Need to rethink this a bit. If IOException occurs when trying to create dir or file (eg because user doesn't have write permission), then what happens? Can we still start the app, but then prompt them for the location (like we do for the backup dir)? If so, that obviously won't work for helm install, although that might not be an issue, because we have more control in k8s over write locations.

(Also note for reference that the Tomcat user sometimes doesn't necessarily even have a home directory)

Reply:
Discussed with @taojing2002 5/16/23: the exception is rethrown up the chain, resulting in the MetacatServlet init() method throwing a ServletException, thus not allowing the application to start - which is probably what we want to happen. We also agreed that I should provide more transparency in the logs around what might have gone wrong - so I've made those changes in commit 88c62e52c1da151a613b817b82a89bb71a0d9d35